serverv2/core/appmanager/app.go

```diff
@@ -0,0 +1,67 @@
+package appmanager
+
+import (
+	"context"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/serverv2/core/event"
+	"github.com/cosmos/cosmos-sdk/serverv2/core/transaction"
+	"github.com/cosmos/cosmos-sdk/serverv2/core/validator"
+)
+
+type App[T transaction.Tx] interface {
+	ChainID() string
+	AppVersion() (uint64, error)
+
+	InitChain(context.Context, RequestInitChain) (ResponseInitChain, error)
+	DeliverBlock(context.Context, RequestDeliverBlock[T]) (ResponseDeliverBlock, error)
+
+	Query(context.Context, *QueryRequest) (*QueryResponse, error)
+}
+
+type QueryRequest struct {
+	Height int64
+	Path   string
+	Data   []byte
+}
+
+type QueryResponse struct {
+	Height int64
+	Value  []byte
```

Comment:
no proof?

Reply:
we dont support range proofs at the moment. @alexanderbez have you put thoughts to this or should we say the future graphql/client stuff would handle this?

Reply:
In store v2, we only support a `CommitmentOp`

* https://github.com/cosmos/cosmos-sdk/blob/main/store/store.go#L180
* https://github.com/cosmos/cosmos-sdk/blob/main/store/proof.go#L16-L28